### ISSUE TITLE
com/intellij/plugins/thrift/jps/ModelSerializerExtension on idea2020.3

### ISSUE THREAD

**Issue by Nasller** [fkorotkov/intellij-thrift#101]
java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0

**Comment by ekzeb** [comment]
> java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0

Project Settings/Project SDK/1.8 java version "1.8.0_271"

**Comment by Nasller** [comment]
> > java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
> 
> Project Settings/Project SDK/1.8 java version "1.8.0_271"

whats mean , using 1.8.0_271 version?

**Comment by ekzeb** [comment]
> > > java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
> > 
> > 
> > Project Settings/Project SDK/1.8 java version "1.8.0_271"
> 
> whats mean , using 1.8.0_271 version?

concrete JDKs version used in Project, if you can, not my case, change it to 11 jdk and you'll not see this error

**Comment by Nasller** [comment]
> > > > java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
> 
> > > 
> 
> > > 
> 
> > > Project Settings/Project SDK/1.8 java version "1.8.0_271"
> 
> > 
> 
> > whats mean , using 1.8.0_271 version?
> 
> 
> 
> concrete JDKs version used in Project, if you can, not my case, change it to 11 jdk and you'll not see this error

i know. but i cant change .thanks

**Comment by fkorotkov** [comment]
I don't think it's related to your Project JDK. The JPS part should use IDE's JDK. Can you share your About IntelliJ screen? Which JDK your IntelliJ is using. 🤔

**Comment by Nasller** [comment]
> I don't think it's related to your Project JDK. The JPS part should use IDE's JDK. Can you share your About IntelliJ screen? Which JDK your IntelliJ is using. 🤔 

i think too. But it actually uses my project jdk.i try to change my project jdk to 11 and thats worked.use jdk11 no error.my intellij jdk used defaults jdk11.

**Comment by yuemingming** [comment]
I also encountered the same problem

**Comment by yuemingming** [comment]
Is there any current solution 

> java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0

**Comment by Nasller** [comment]
> Is there any current solution
> 
> > java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0

disabled the plugin

**Comment by TKYAngus** [comment]
I got the same issue with my project JDK 8 on IntelliJ IDEA 2020.3.
I have to disable the plugin to make it compile.

**Comment by cny-dixa** [comment]
I also have the issue.
The problem is that I cannot change to Java 11 since my company develops in a Scala version which requires Java 8.

**Comment by Nasller** [comment]
Everys guys, I found a temporary solution. Remove the files in the `META-INF.services` file directory in `thrift-jps.jar`, so that you can temporarily use the function of thrift without reporting an build error. I guess it may affect To the function of the thrift compiler, but that function seems useless.Hope the author can fix it.Only applies to me, hope it can help you.

**Comment by guynaa** [comment]
> Everys guys, I found a temporary solution. Remove the files in the `META-INF.services` file directory in `thrift-jps.jar`, so that you can temporarily use the function of thrift without reporting an build error. I guess it may affect To the function of the thrift compiler, but that function seems useless.Hope the author can fix it.Only applies to me, hope it can help you.

thanks, workaround is working for me.

**Comment by binbjz** [comment]
The same issues.
```
Internal error (java.lang.UnsupportedClassVersionError): com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
java.lang.UnsupportedClassVersionError: com/intellij/plugins/thrift/jps/ModelSerializerExtension has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

**Comment by fkorotkov** [comment]
Just published 1.5.0 where JPS plugins should be Java 8 compatible. Let's see if it fixes the issue.
